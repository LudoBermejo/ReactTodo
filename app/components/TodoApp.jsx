import uuid from 'node-uuid';
import React from 'react';

import TodoList from 'TodoList';
import AddTodoItem from 'AddTodoItem';
import TodosFilter from 'TodosFilter';
import TodoAPI from 'TodoAPI';

export default class TodoApp extends React.Component {
  constructor(props) {
    super(props);

    this.addTodoItem = this.addTodoItem.bind(this);
    this.search = this.search.bind(this);
    this.changeCompleteTodo = this.changeCompleteTodo.bind(this);
    this.state = {
      searchText: '',
      filterByComplete: false,
      todoList: TodoAPI.getTodos()
    };
  }

  componentDidUpdate() {
    TodoAPI.setTodos(this.state.todoList);
  }

  search(value) {
    this.setState({ value });
  }

  addTodoItem(value) {
    const todoList = this.state.todoList;

    this.setState({
      todoList: [
        ...todoList,
        {
          id: uuid(),
          value,
          completed: false
        }
      ]
    });
  }

  changeCompleteTodo(id) {
    this.setState({
      todoList: this.state.todoList.map((item) => {
        const todo = item;
        if (todo.id === id) {
          todo.completed = !item.completed;
        }
        return todo;
      })
    });
  }

  render() {
    const { todoList } = this.state;
    return (
      <div>
        <h1>Todo</h1>
        <TodosFilter onSearch={this.search} />
        <TodoList todoList={todoList} onChangeItemComplete={this.changeCompleteTodo} />
        <AddTodoItem onAddItem={this.addTodoItem} />
      </div>
    );
  }
}

