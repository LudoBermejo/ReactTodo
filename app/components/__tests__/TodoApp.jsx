import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import TodoApp from 'TodoApp';

describe('Todo app', function() {

  it('renders properly', function() {
    const todoApp = renderer.create(<TodoApp />).toJSON();
    expect(todoApp).toMatchSnapshot();
  });

})

