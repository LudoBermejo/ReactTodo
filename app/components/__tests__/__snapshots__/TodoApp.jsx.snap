exports[`Todo app renders properly 1`] = `
<div>
  <h1>
    Todo
  </h1>
  <div>
    <div>
      <input
        onChange={[Function]}
        placeholder="Search in todos"
        type="search" />
    </div>
    <label>
      <input
        onChange={[Function]}
        type="checkbox" />
    </label>
  </div>
  <div>
    <h1>
      List of items:
    </h1>
    <ul>
      <li
        id="list_item_e8960427-2ae4-44c6-b1f2-361571f4b883"
        onClick={[Function]}>
        <input
          checked={false}
          type="checkbox" />
        first
      </li>
      <li
        id="list_item_a6b5973f-e16e-4fc8-85ae-1365635aa96f"
        onClick={[Function]}>
        <input
          checked={false}
          type="checkbox" />
        Second
      </li>
      <li
        id="list_item_17249fa1-b844-499b-8b3b-a781911f841b"
        onClick={[Function]}>
        <input
          checked={false}
          type="checkbox" />
        Third
      </li>
      <li
        id="list_item_01dbcb1b-d9f6-478c-bda2-092db73b3a80"
        onClick={[Function]}>
        <input
          checked={false}
          type="checkbox" />
        Fourth
      </li>
    </ul>
  </div>
  <form
    className="form"
    onSubmit={[Function]}>
    <input
      placeholder="What do you need to do?"
      type="text" />
    <button
      className="button expanded">
      Add todo
    </button>
  </form>
</div>
`;
